significant_strikes.py

```diff
@@ -0,0 +1,500 @@
+        
+
+    def get_raw_significant_strikes_stats(self, df, round):
+        """
+        Extracts raw significant strike stats from the dataframe for a specific round.
+
+        Args:
+            df (pd.DataFrame): The dataframe containing fight records.
+            round (str): The round for which the statistics are required.
+
+        Returns:
+            pd.DataFrame: A dataframe with raw stats for the specified round for fighters a and b.
+        """
+
+        # Define column names based on the round
+        if round == "overall":
+            # Use overall stats columns
+            sig_str_cols = [
+                'fighter_a_total_sig_str_landed',
+                'fighter_a_total_sig_str_attempted',
+                'fighter_b_total_sig_str_landed',
+                'fighter_b_total_sig_str_attempted'
+            ]
+        else:
+            # Use specific round stats columns
+            sig_str_cols = [
+                f'fighter_a_round_{round[1]}_sig_str_landed',
+                f'fighter_a_round_{round[1]}_sig_str_attempted',
+                f'fighter_b_round_{round[1]}_sig_str_landed',
+                f'fighter_b_round_{round[1]}_sig_str_attempted'
+            ]
+
+        # Initialize an empty DataFrame with the specified columns
+        stats_df = pd.DataFrame(columns=[
+            'total_fight_time', 'fighter_a_id', 'fighter_b_id',
+            'fighter_a_sig_str_landed', 'fighter_a_sig_str_attempted',
+            'fighter_b_sig_str_landed', 'fighter_b_sig_str_attempted'
+        ])
+
+        # Iterate through each fight in the DataFrame
+        for index, fight in df.iterrows():
+            # Calculate the total fight time
+            total_fight_time = self.get_round_time(fight['outcome_round'], fight['outcome_time'], round)
+
+            # Extract relevant columns for the round
+            sig_str_stats = [fight[col] for col in sig_str_cols]
+
+            # Create a list of stats for the current fight
+            fight_stats = [
+                total_fight_time, fight['fighter_a_id'], fight['fighter_b_id']
+            ] + sig_str_stats
+
+            # Append the stats to the DataFrame
+            stats_df.loc[index] = fight_stats
+
+        return stats_df
+
+    def calculate_significant_strikes_landed(self, stats_by_fight_df, fighter_id):
+        """
+        Calculates the rate of significant strikes landed per minute by a fighter.
+
+        Args:
+            stats_by_fight_df (pd.DataFrame): The dataframe containing aggregated fight stats.
+            fighter_id (str): The ID of the fighter.
+
+        Returns:
+            float: The rate of significant strikes landed per minute in the given fights.
+        """
+
+        # Initialize variables to store totals
+        sig_str_landed = 0
+        total_fight_time = 0
+
+        # Iterate through each fight in the DataFrame
+        for index, fight in stats_by_fight_df.iterrows():
```

Comment:
You can use the .sum() function to sum up columns. In this case it is a little tricky, but you can do something like this:

`fighter_a_id_vals = stats_by_fight_df.fighter_a_id.values`
`fighter_b_id_vals = stats_by_fight_df.fighter_b_id.values`
`sig_str_landed = stats_by_fight_df[fighter_a_id_vals == fighter_id]['fighter_a_sig_str_landed'].sum() + stats_by_fight_df[fighter_b_id_vals == fighter_id]['fighter_b_sig_str_landed'].sum()`

This will vectorize the operation and make it MUCH faster than doing a for loop. 
See lines 53-57 [here](https://github.com/ivan-guo/OctaPicks/pull/33/commits/eb2ee116f23f5cac5c4668735009b88385c99f01).

Reply:
ok bet imma apply this to the other funcs that do a similar thing